### PR TITLE
Sonar integration

### PR DESCRIPTION
**Pull request by Aidasvenc**
Set up a SonarCloud integration for CI code checks.

### CODE REVIEW

**Comment by Aidasvenc** [comment]
Finally it works 🥇

**Comment by Aidasvenc** [comment]
Will need to set up code coverage analysis when we set up a testing library though.

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Build
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (AlbertoCentonze) [inline .github/workflows/sonar.yml:7]: Do we need this on `reopened` as well? When one is reopening a PR (if ever) I would assume there's not any new commit to analyze.

File: .github/workflows/sonar.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (AlbertoCentonze) [inline .github/workflows/sonar.yml:13]: Please update this to the latest version of checkout (v4)

**Review by AlbertoCentonze — CHANGES_REQUESTED** [review]
Nice job! We'll probably need to customize the settings a bit more as we use it but for now it's a great start.

File: .github/workflows/sonar.yml (hunk 1 above)
Comment (Aidasvenc) [reply .github/workflows/sonar.yml:7]: Copied from the suggested code in the docs. You are right.

**Review by AlbertoCentonze — APPROVED** [review]
LGTM

**Pull request merged**